sdcm/nemesis.py

```diff
@@ -1088,6 +1089,68 @@ def disrupt_ldap_connection_toggle(self):
+        4. Create a new super-user role in Scylla.
+        5. Create a new super-user role in Ldap, with the new user as a member.
+        6. Verify this new user is authorised on resources.
+        7. Modify Ldap role to remove membership of new user.
+        8. Verify the new user is unauthorised on resources (create a table).
+        9. Do clean-up: delete Ldap Role, Scylla roles and keyspace/tables.
+
+        """
+        if not self.cluster.params.get('use_ldap_authorization'):
+            raise UnsupportedNemesis('Cluster is not configured to run with LDAP authorization, hence skipping')
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis('Cluster is not enterprise. LDAP is supported only for enterprise. Skipping')
+        node = self.cluster.nodes[0]
+        superuser_role = 'superuser_role'
+        new_test_user = 'new_test_user'
+        self.log.debug("Create new user in Scylla")
+        self.tester.create_role_in_scylla(node=node, role_name=new_test_user, is_superuser=False,
+                                          is_login=True)
+        with pytest.raises(Unauthorized, match="has no CREATE permission"):
+            with self.cluster.cql_connection_patient(node=node, user=new_test_user, password=LDAP_PASSWORD) as session:
+                session.execute(
+                    """ CREATE KEYSPACE IF NOT EXISTS customer WITH replication = {'class': 'SimpleStrategy',
+                    'replication_factor': 1} """)
+
+        self.log.debug("Create a new super-user role in Scylla")
+        self.tester.create_role_in_scylla(node=node, role_name=superuser_role, is_superuser=True,
+                                          is_login=False)
+        self.cluster.wait_for_schema_agreement()
+        self.log.debug("Create a new super-user role in Ldap, associated with the new user")
+        if self.cluster.params.get('prepare_saslauthd'):
+            self.tester.add_user_in_ldap(username=new_test_user)
+        self.tester.create_role_in_ldap(ldap_role_name=superuser_role, unique_members=[new_test_user, LDAP_USERS[1]])
+
+        self.tester.wait_for_user_roles_update(are_roles_expected=True, username=new_test_user)
+        self.log.debug("Create keyspace and table where authorized")
+        self.tester.wait_verify_user_permissions(username=new_test_user)
+
+        self.log.debug("Remove authorization and verify unauthorized user")
+        self.tester.modify_ldap_role_delete_member(ldap_role_name=superuser_role, member_name=new_test_user)
+        self.tester.wait_for_user_roles_update(are_roles_expected=False, username=new_test_user)
+        self.tester.wait_verify_user_no_permissions(username=new_test_user)
+
+        # Clean-up resources
+        self.tester.delete_ldap_role(ldap_role_name=superuser_role)
+        if self.cluster.params.get('prepare_saslauthd'):
+            self.tester.delete_ldap_role(ldap_role_name=new_test_user)
```

Comment:
AFAICT it won't work on `MS AD`, but perhaps it will... did you test it?

Reply:
Tested in:
https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-100gb-4h-ldap-authorization-authenticator-test-yg/56/

Reply:
There might be a general existing issue, using MS AD in SCT.
The test failed with:
```
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > Traceback (most recent call last):
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3911, in wrapper
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     result = method(*args[1:], **kwargs)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1129, in disrupt_ldap_role_add_remove_permission
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     self.tester.create_role_in_ldap(ldap_role_name=superuser_role, unique_members=[new_test_user, LDAP_USERS[1]])
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 558, in create_role_in_ldap
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     self._add_ldap_entry(ldap_entry=role_entry)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 545, in _add_ldap_entry
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     self.localhost.add_ldap_entry(ip=self.ldap_ip, ldap_port=self.ldap_port,
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/ldap.py", line 82, in add_ldap_entry
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     LdapContainerMixin.create_ldap_connection(ip, ldap_port, user, password)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/decorators.py", line 67, in inner
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     return func(*args, **kwargs)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/ldap.py", line 69, in create_ldap_connection
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     LdapContainerMixin.ldap_server = Server(host=f'ldap://{ip}:{ldap_port}', get_info=ALL)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/usr/local/lib/python3.10/site-packages/ldap3/core/server.py", line 118, in __init__
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     raise LDAPInvalidPortError('port must be an integer')
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > ldap3.core.exceptions.LDAPInvalidPortError: port must be an integer
```
I think we can leave this mode out of this PR content and address it in a future PR.
Currently skipping this mode by:
```
        if not self.cluster.params.get('ldap_server_type') == LdapServerType.MS_AD:
            raise UnsupportedNemesis('Cluster is not configured to run with open LDAP authorization, hence skipping')
```